# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Refresh / Content Opportunity Scoring

I chose this lane because it combines machine learning with a practical ranking problem. The objective is to identify which content pages should be reviewed first for refresh based on observable search and engagement signals. This project involves feature engineering, classification, ranking, and explainable recommendations, making it a strong end-to-end ML project.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

My decisions will improve the 

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print(df.shape)
print(df["trend_direction"].value_counts())
print(df["impressions_90d"].describe())

(30000, 44)
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64
count     30000.000000
mean       5200.366300
std       16838.019547
min           1.000000
25%          81.000000
50%         731.000000
75%        3615.250000
max      517715.000000
Name: impressions_90d, dtype: float64


### Initial Observations

- The starter dataset contains **30,000 content pages** and **44 features**, providing enough data for meaningful exploratory analysis and model development.

- The target distribution shows that **16,262 pages (54.2%)** are currently marked as **"down"**, suggesting that declining content is common and that prioritizing refresh opportunities could provide significant value.

- The `impressions_90d` feature is highly skewed:
  - Median impressions: **731**
  - Mean impressions: **5,200**
  - Maximum impressions: **517,715**

  This indicates that while most pages receive relatively low traffic, a small number of pages attract extremely high visibility. These high-impact pages should be prioritized carefully because poor recommendations could affect a large amount of search traffic.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What my work can claim

- This project identifies content pages that are likely to benefit from review based on observable search and engagement signals.
- The model provides a ranked list of pages along with transparent reason codes to support content review decisions.
- The results are decision-support recommendations derived from historical data patterns rather than guarantees.
- The analysis measures relationships between observed features and content performance within the available dataset.

### What my work cannot claim

- This project cannot prove that refreshing a page will cause an increase in search traffic or rankings.
- It cannot predict or reverse-engineer Google's ranking algorithm.
- It cannot establish causal relationships between content changes and future performance without controlled experiments.
- The recommendations should be reviewed by human analysts before any content decisions are made.

## Self-check

Before you submit, confirm each line honestly:

- [✓] Every section above is filled — markdown thinking AND the code that backs it
- [✓] The notebook runs top to bottom with no errors (Runtime → Run all)
- [✓] No client names, URLs, or private queries anywhere
- [✓] My claims use careful words: observed, measured, directional, decision-support
- [✓] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.